carapace-server/src/main/java/org/carapaceproxy/api/CertificatesResource.java

```diff
@@ -261,25 +269,40 @@ public Response createCertificate(CertificateForm form) {
         return FormValidationResponse.created();
     }
 
+    @DELETE
+    @Path("{certId}")
+    public Response createCertificate(@PathParam("certId") final String certId) {
+        final var server = (HttpProxyServer) context.getAttribute("server");
+        final var certificates = server.getCurrentConfiguration().getCertificates();
+        if (!certificates.containsKey(certId)) {
+            throw new WebApplicationException(Response.Status.NOT_FOUND);
+        }
+        try {
+            server.rewriteConfiguration(it -> it.removeCertificate(certId));
+            return SimpleResponse.ok();
+        } catch (ConfigurationChangeInProgressException | InterruptedException | ConfigurationNotValidException e) {
+            return SimpleResponse.error(e);
+        }
+    }
+
     @GET
     @Produces(MediaType.APPLICATION_OCTET_STREAM)
     @Path("{certId}/download")
-    public Response downloadCertificateById(@PathParam("certId") String certId) throws GeneralSecurityException {
+    public Response downloadCertificateById(@PathParam("certId") final String certId) {
         CertificateBean cert = findCertificateById(certId);
-        byte[] data = new byte[0];
-        if (cert != null && cert.isDynamic()) {
-            HttpProxyServer server = (HttpProxyServer) context.getAttribute("server");
-            DynamicCertificatesManager dynamicCertificateManager = server.getDynamicCertificatesManager();
-            data = dynamicCertificateManager.getCertificateForDomain(cert.getId());
+        if (cert == null || !cert.isDynamic()) {
```

Comment:
This is a behaviour change. I'm ok with this but next time It would be best to create another PR.

We must remember to write this change in the release notes

Reply:
Actually, it only changed the error from a non-meaningful 500 originated from a NullPointerException to a more meaningful 404.

But you're right, I should've stated the updated somewhere